### ISSUE TITLE
2.6 -> 2.7 behavior change when port=None is explicitly passed to connect

### ISSUE THREAD

**Issue by reece** [psycopg/psycopg2#524]
With 2.6.2, `connect()` would use the default pg port if no `port` argument was passed or if an explicit `port=None` was passed. In 2.7, the latter results in an error.

## 2.6.2
```
In [1]: import psycopg2

In [2]: psycopg2.__version__
Out[2]: '2.6.2 (dt dec pq3 ext lo64)'

In [3]: c = psycopg2.connect(host="uta.biocommons.org", user="anonymous", password="", dbname="uta")

In [4]: c = psycopg2.connect(host="uta.biocommons.org", user="anonymous", password="", dbname="uta", port=None)

```

## 2.7
```
In [1]: import psycopg2

In [2]: psycopg2.__version__
Out[2]: '2.7 (dt dec pq3 ext lo64)'

In [3]: c = psycopg2.connect(host="uta.biocommons.org", user="anonymous", password="", dbname="uta")

In [4]: c = psycopg2.connect(host="uta.biocommons.org", user="anonymous", password="", dbname="uta", port=None)
---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)
<ipython-input-4-c6225dce1768> in <module>()
----> 1 c = psycopg2.connect(host="uta.biocommons.org", user="anonymous", password="", dbname="uta", port=None)

/home/reece/projects/biocommons/hgvs/venv/local/lib/python2.7/site-packages/psycopg2/__init__.pyc in connect(dsn, connection_factory, cursor_factory, **kwargs)
    128 
    129     dsn = _ext.make_dsn(dsn, **kwargs)
--> 130     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
    131     if cursor_factory is not None:
    132         conn.cursor_factory = cursor_factory

OperationalError: invalid port number: "None"

```

**Comment by reece** [comment]
P.S. Seems to me that the culprit is that these None checks were removed in 2.7:

https://github.com/psycopg/psycopg2/blob/1690fbace1bf498abd5a4c4f68bfaab609bc3587/lib/__init__.py#L147 (from 2.6.2)

**Comment by dvarrazzo** [comment]
Yes, it's already been fixed, see #517. To be released in 2.7.1.
